### ISSUE TITLE
NiftiMapsMasker and NiftiLabelsMasker cannot transform 3D images

### ISSUE THREAD

**Issue by jeromedockes** [nilearn/nilearn#2637]
documentation suggests they can (says "imgs: 3D/4D Niimg-like object").
example:
```
from nilearn import datasets
from nilearn.input_data import NiftiMapsMasker, NiftiLabelsMasker


img = datasets.load_mni152_template()

atlas_img = datasets.fetch_atlas_destrieux_2009()["maps"]
labels_masker = NiftiLabelsMasker(atlas_img)

prob_atlas_img = datasets.fetch_atlas_harvard_oxford("cort-prob-2mm")["maps"]
maps_masker = NiftiMapsMasker(prob_atlas_img)

# OK
labels_masker.fit_transform([img])
maps_masker.fit_transform([img])

# both lines below fail
labels_masker.fit_transform(img)
maps_masker.fit_transform(img)
```

**Comment by jeromedockes** [comment]
fit on 3d images is always fine, fit_transform and transform fail. more details:
```
import itertools

from sklearn.base import clone

from nilearn import datasets
from nilearn.input_data import NiftiMapsMasker, NiftiLabelsMasker


img_3d = datasets.load_mni152_template()
images = {"3d": img_3d, "4d": [img_3d]}

atlas_img = datasets.fetch_atlas_destrieux_2009()["maps"]
labels_masker = NiftiLabelsMasker(atlas_img)

prob_atlas_img = datasets.fetch_atlas_harvard_oxford("cort-prob-2mm")["maps"]
maps_masker = NiftiMapsMasker(prob_atlas_img)

for masker, img in itertools.product(
    [labels_masker, maps_masker], images.keys()
):
    masker_name = masker.__class__.__name__

    masker = clone(masker)
    print(f"{masker_name} fit on {img}: ", end="")
    try:
        masker.fit(images[img])
        print("OK")
    except Exception:
        print("Fail")

    masker = clone(masker)
    print(f"{masker_name} fit_transform on {img}: ", end="")
    try:
        masker.fit_transform(images[img])
        print("OK")
    except Exception:
        print("Fail")

    masker = clone(masker)
    masker.fit(images["4d"])
    print(f"{masker_name} transform {img}: ", end="")
    try:
        masker.transform(images[img])
        print("OK")
    except Exception:
        print("Fail")
```

**Comment by jeromedockes** [comment]
see also https://neurostars.org/t/extract-atlas-based-values-from-a-3d-niiimg-object/17730

**Comment by bthirion** [comment]
I guess that the most idiomatic fix is to  transform the 3D img to a list in the call to transform.

**Comment by jeromedockes** [comment]
yes I think so!

**Comment by thomasbazeille** [comment]
add if anyone is interested in tackling this PR, the additional requirements would be :

- [ ] Add a test that fails with current version and would be solved by PR (basically make tests on both maskers a bit more extensive)

**Comment by NicolasGensollen** [comment]
@achamma723 I think this could be a good first issue for you. Let me know if something isn't clear!

**Comment by NicolasGensollen** [comment]
Addressed in #2708
